activesupport/lib/active_support/evented_file_update_checker.rb

```diff
@@ -96,15 +67,59 @@ def execute_if_updated
       end
     end
 
-    private
-      def boot!
+    class Core
+      attr_reader :updated
+
+      def initialize(files, dirs)
+        @files = files.map { |file| Pathname(file).expand_path }.to_set
+
+        @dirs = dirs.each_with_object({}) do |(dir, exts), hash|
+          hash[Pathname(dir).expand_path] = Array(exts).map { |ext| ext.to_s.sub(/\A\.?/, ".") }.to_set
+        end
+
+        @common_path = @dirs.keys.reduce { |path1, path2| common_path(path1, path2) }
+
+        @dtw = directories_to_watch
+        @missing = []
+
+        @updated = Concurrent::AtomicBoolean.new(false)
+        @mutex = Mutex.new
+
+        start
+        @after_fork = ActiveSupport::ForkTracker.after_fork { start }
+      end
+
+      def finalizer
+        proc do
+          stop
+          ActiveSupport::ForkTracker.unregister(@after_fork)
```

Comment:
What about ActiveSupport::ProcessForking as the name here?

Reply:
`ActiveSupport::ForkTracker` was actually introduced by #37312.  I too was pleasantly surprised to find it was already written! :smile:  It hasn't been publicly released yet though, so I can look into renaming it in a separate PR if you would like.